app/src/main/java/space/taran/arkmemo/data/repositories/TextNotesRepository.kt

```diff
@@ -17,69 +20,156 @@ import kotlin.io.path.getLastModifiedTime
-                            val meta = ResourceMeta(
-                                id,
-                                filePath.fileName.toString(),
-                                filePath.extension,
-                                filePath.getLastModifiedTime(),
-                                size
-                            )
-
-                            val note = TextNote(content, meta)
-                            notes.add(note)
-                            close()
-                        }
+                        val fileName = filePath.fileName.toString()
+                        val rootResourceId: ResourceId = fileName.toLong()
+                        val meta = VersionMeta(
+                            rootResourceId
+                        )
+                        val versionStorage = VersionStorage(getPath()!!)
+                        val verContent = Version.Content(versionStorage.versions(rootResourceId))
+                        val ver = Version(verContent, meta)
+                        versions.add(ver)
                     } catch (e: Exception) {
                         e.printStackTrace()
                     }
                 }
             }
         }
-        return notes
+        return versions
     }
 
-    private fun createTextNoteFile(path: Path?, noteString: String?) {
+    private fun getLastChildrenId(rootResourceId: ResourceId): ResourceId {
+        val path = getPath()
+        var lastChildResourceId: ResourceId? = null
+        if (path != null) {
+            val versionDir = File(path.toFile(), ARK_VERSIONS_DIR)
+            val versionFile = File(versionDir, rootResourceId.toString())
+            if (versionFile.exists()) {
+                val fileReader = FileReader(versionFile)
+                val bufferedReader = BufferedReader(fileReader)
+                val jsonVersion = StringBuilder()
+                with(bufferedReader) {
+                    forEachLine {
+                        jsonVersion.append(it)
+                    }
+                }
+                lastChildResourceId = if (jsonVersion.toString() == "") {
+                    rootResourceId
+                } else {
+                    val versionContent =
+                        JsonParser.parseVersionContentFromJson(jsonVersion.toString())
+                    if (versionContent.idList.isEmpty()) {
+                        rootResourceId
+                    } else {
+                        versionContent.idList.last()
+                    }
+                }
+            }
+        }
+        return lastChildResourceId!!
+    }
+
+    private fun getNote(ResourceId: ResourceId): TextNote {
+        val path = getPath()
+        var note: TextNote? = null
+        if (path != null) {
+            val noteFile = File(path.toFile(), "$ResourceId.$NOTE_EXT")
+            val notePath = noteFile.toPath()
+            try {
+                val fileReader = FileReader(noteFile)
```

Comment:
I think we should extract common code working with files. Would be easier to read.